packages/core/src/sdk/overrides/PageProvider.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  ClientProductGalleryQueryQuery,
+  ClientProductQueryQuery,
+  ClientProductsQueryQuery,
+  ServerCollectionPageQueryQuery,
+  ServerProductPageQueryQuery,
+} from '@generated/graphql'
+import type { PropsWithChildren } from 'react'
+import { createContext, useContext, useMemo } from 'react'
+import { SearchPageContextType } from 'src/pages/s'
+
+export interface ProductDetailsPageContext {
+  data?: ServerProductPageQueryQuery &
+    ClientProductQueryQuery['product'] & { isValidating?: boolean }
+}
+
+export type ProductsPerPage = {
+  page: number
+  data: ClientProductsQueryQuery
+}
+
+export interface ProductListingPageContext {
+  data?: ServerCollectionPageQueryQuery &
+    ClientProductGalleryQueryQuery & { productsPerPage: ProductsPerPage[] }
+}
+
+export interface SearchPageContext {
+  data?: SearchPageContextType &
+    ClientProductGalleryQueryQuery & { productsPerPage: ProductsPerPage[] }
+}
+
+export const isPDP = (x: any): x is ProductDetailsPageContext =>
+  x?.data?.product?.sku != undefined && x?.data?.product?.sku != null
+
+export const isPLP = (x: any): x is ProductListingPageContext =>
+  x?.data?.collection?.seo != undefined &&
+  x?.data?.collection?.seo != null &&
+  x?.data?.collection?.sku == undefined
+
+export const isSearchPage = (x: any): x is SearchPageContext =>
```

Comment:
I get why these things are treated separately but if we want to keep it that way I think we need to change the PLP name. AFAIK the concept of product listing pages includes search pages. One may use the PLP context thinking it might apply to search pages as well.

Reply:
Do you have name suggestions? I think we are going to simplify this understandings and handle this using the documentation. WDYT?

Reply:
Isn't the SearchPage a specification of a PLP? Meaning that a SearchPage **is** a PLP but needs to have additional things checked (presence of `searchTerm`)? I think we can leave the differentiation here.

Reply:
I liked @gvc mindset. I think we can go for it.